### PR TITLE
AK: Add Vector formatter

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
For debugging purposes, it is very useful to look at a Vector in a
simple list representation. Therefore, the new Formatter for Vector
provides a string representation of the following form:

```
[ 1, 2, 3, 4, 5 ]
```

This requires the content type of Vector to be formattable with default
arguments.

The current implementation ignores width and precision, which may be
accounted for later or passed down to the content formatter.

### CODE REVIEW

File: Userland/Libraries/LibAudio/FlacLoader.cpp [hunk 1]
```diff
@@ -639,15 +639,7 @@ Vector<i32> FlacLoaderPlugin::decode_custom_lpc(FlacSubframeHeader& subframe, In
...
-    if constexpr (AFLACLOADER_DEBUG) {
-        StringBuilder coefficients_formatted;
-        coefficients_formatted.append("[ ");
-        for (auto coeff : coefficients) {
-            coefficients_formatted.append(String::formatted("{}, ", coeff));
-        }
-        coefficients_formatted.append("]");
-        dbgln("{}-bit {} shift coefficients: {}", lpc_precision, lpc_shift, coefficients_formatted.to_string());
-    }
+    dbgln_if(AFLACLOADER_DEBUG, "{}-bit {} shift coefficients: {}", lpc_precision, lpc_shift, coefficients);
```
Comment (alimpfard) [inline Userland/Libraries/LibAudio/FlacLoader.cpp:642]: Let's move this into another commit, it's not very related to AK afterall :)

**Comment by alimpfard** [comment]
:thinking:
Perhaps the first commit can be reworded a little better (e.g. "Add formatter for Vector"?)

**Comment by kleinesfilmroellchen** [comment]
Ok, I'll do that later
